pkgs/build-support/dotnet/make-nuget-deps/default.nix

```diff
@@ -1,23 +1,43 @@
-{ symlinkJoin
-, fetchurl
-, stdenvNoCC
-, lib
-, unzip
-, patchNupkgs
-, nugetPackageHook
-, fetchNupkg
+{
+  symlinkJoin,
+  lib,
+  fetchNupkg,
 }:
-lib.makeOverridable(
-  { name
-  , nugetDeps ? import sourceFile
-  , sourceFile ? null
-  , installable ? false
+lib.makeOverridable (
+  {
+    name,
+    nugetDeps ? null,
+    sourceFile ? null,
+    installable ? false,
   }:
+  let
+    callNupkg = pkg: if lib.isDerivation pkg then pkg else fetchNupkg pkg;
+
+    loadDeps =
+      x:
```

Comment:
Feels a little messy that we need these functions defined in both add-nuget-deps and make-nuget-deps. I wonder if there's a sensible way to refactor this so that both use a common implementation?

Reply:
I have a feeling `mkNugetDeps` and `addNugetDeps` might be redundant but I'm not really sure, I think I'll leave it like this for now since it's such a small snippet but possibly look into merging them together or something.